runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonScript.java

```diff
@@ -473,6 +476,89 @@ private void teleportToFeroxEnclave() {
         logOnceToChat("No Ring of Dueling found in inventory for teleporting to Ferox Enclave.");
     }
 
+    private void evaluateAndConsumePotions(TormentedDemonConfig config) {
+        int currentStrength = Microbot.getClient().getBoostedSkillLevel(Skill.STRENGTH);
+        int maxStrength = Microbot.getClient().getRealSkillLevel(Skill.STRENGTH);
+        int currentRange = Microbot.getClient().getBoostedSkillLevel(Skill.RANGED);
+        int maxRange = Microbot.getClient().getRealSkillLevel(Skill.RANGED);
+
+        int threshold = config.boostedStatsThreshold();
+
+        List<Integer> combatPotionIds = null;
+        switch (config.combatPotionType()) {
+            case SUPER_COMBAT:
+                combatPotionIds = List.of(
+                        ItemID.SUPER_COMBAT_POTION4,
+                        ItemID.SUPER_COMBAT_POTION3,
+                        ItemID.SUPER_COMBAT_POTION2,
+                        ItemID.SUPER_COMBAT_POTION1
+                );
+                break;
+            case DIVINE_SUPER_COMBAT:
+                combatPotionIds = List.of(
+                        ItemID.DIVINE_SUPER_COMBAT_POTION4,
+                        ItemID.DIVINE_SUPER_COMBAT_POTION3,
+                        ItemID.DIVINE_SUPER_COMBAT_POTION2,
+                        ItemID.DIVINE_SUPER_COMBAT_POTION1
+                );
+                break;
+        }
+
+        List<Integer> rangingPotionIds = null;
+        switch (config.rangingPotionType()) {
+            case RANGING:
+                rangingPotionIds = List.of(
+                        ItemID.RANGING_POTION4,
+                        ItemID.RANGING_POTION3,
+                        ItemID.RANGING_POTION2,
+                        ItemID.RANGING_POTION1
+                );
+                break;
+            case DIVINE_RANGING:
+                rangingPotionIds = List.of(
+                        ItemID.DIVINE_RANGING_POTION4,
+                        ItemID.DIVINE_RANGING_POTION3,
+                        ItemID.DIVINE_RANGING_POTION2,
+                        ItemID.DIVINE_RANGING_POTION1
+                );
+                break;
+        }
+
+        if (combatPotionIds != null && getPercentageDifference(currentStrength, maxStrength) < threshold) {
+            Rs2Item combatPotion = findPotionInInventory(combatPotionIds);
+            if (combatPotion != null) {
+                Rs2Inventory.interact(combatPotion, "Drink");
+                logOnceToChat("Drinking combat potion: " + combatPotion.getName());
+            } else {
+                logOnceToChat("No combat potions left in inventory.");
+            }
+        }
+
+        if (rangingPotionIds != null && getPercentageDifference(currentRange, maxRange) < threshold) {
```

Comment:
Rs2Player also has a bunch of premade methods to check if  divine bastion is still active

```
    public static boolean hasRangingPotionActive() {
        return Microbot.getClient().getBoostedSkillLevel(Skill.RANGED) - 5 > Microbot.getClient().getRealSkillLevel(Skill.RANGED);
    }

    public static boolean hasDivineBastionActive() {
        return divineBastionTime > 0;
    }
```

You should check if you can use any of these methods or maybe slightly adjust them to your needs so we don't have the logic for handling potions spread out in each script.

Reply:
Should be fine now